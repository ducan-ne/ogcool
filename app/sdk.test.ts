import { expect, test } from "bun:test"
import ogcool from "@/app/sdk"

test("sdk", () => {
  expect(
    ogcool("UnJS", {
      modifications: [
        {
          name: "Description",
          text: "Hello world",
        },
      ],
    }),
  ).toMatchSnapshot()
})

test("sdk:telemetry-disabled", () => {
  expect(
    ogcool("UnJS", {
      modifications: [
        {
          name: "Description",
          text: "Hello world",
        },
      ],
      disableTelemetry: true,
    }),
  ).toMatchSnapshot()
})

test("sdk:empty", () => {
  expect(ogcool("UnJS")).toMatchSnapshot()
})

// biome-ignore lint/correctness/noUnusedVariables: <explanation>
function testType() {
  ogcool("Beta update x", {
    modifications: [
      {
        name: "Author name",
      },
    ],
  })
}
