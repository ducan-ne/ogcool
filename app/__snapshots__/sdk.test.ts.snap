// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`sdk 1`] = `"http://localhost:3000/templates/v1/tpl_jNvsOYr0cr?d=eyJtb2RpZmljYXRpb25zIjpbeyJuYW1lIjoiRGVzY3JpcHRpb24iLCJ0ZXh0IjoiSGVsbG8gd29ybGQifV19&sdk=ogcool%400.1.6"`;

exports[`sdk:telemetry-disabled 1`] = `"http://localhost:3000/templates/v1/tpl_jNvsOYr0cr?d=eyJtb2RpZmljYXRpb25zIjpbeyJuYW1lIjoiRGVzY3JpcHRpb24iLCJ0ZXh0IjoiSGVsbG8gd29ybGQifV19"`;

exports[`sdk:empty 1`] = `"http://localhost:3000/templates/v1/tpl_jNvsOYr0cr?sdk=ogcool%400.1.6"`;
